src/components/textarea/textarea.test.jsx

```diff
@@ -0,0 +1,227 @@
+    });
+  });
+
+  describe('dirty API', () => {
+    it('is unset without any changes', () => {
+      const ref = createRef();
+      render(<Textarea {...requiredProps} ref={ref} />);
+      expect(ref.current.dirty).toBe(false);
+    });
+
+    it('is set with any changes', () => {
+      const ref = createRef();
+      render(<Textarea {...requiredProps} ref={ref} />);
+      userEvent.type(screen.getByLabelText('the label'), 'a');
+      expect(ref.current.dirty).toBe(true);
+    });
+
+    it('is unset with changes back to initial state', () => {
+      const ref = createRef();
+      render(<Textarea {...requiredProps} ref={ref} />);
+      userEvent.type(screen.getByLabelText('the label'), 'a');
+      userEvent.type(screen.getByLabelText('the label'), '{backspace}');
+      expect(ref.current.dirty).toBe(false);
+    });
+
+    it('is unset with changes and new value prop', () => {
+      const ref = createRef();
+      const { rerender } = render(<Textarea {...requiredProps} ref={ref} />);
+      userEvent.type(screen.getByLabelText('the label'), 'a');
+      rerender(<Textarea {...requiredProps} ref={ref} value="a" />);
+      expect(ref.current.dirty).toBe(false);
+    });
+  });
+
+  describe('id API', () => {
+    it('sets the ID', () => {
+      render(<Textarea {...requiredProps} id="test-id" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('id', 'test-id');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Textarea {...requiredProps} id="unique-id" ref={ref} />);
+      expect(ref.current.id).toBe('unique-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('can be set', () => {
+      render(<Textarea {...requiredProps} label="Unique label" />);
+      expect(screen.getByLabelText('Unique label')).toBeInTheDocument();
+    });
+  });
+
+  describe('name API', () => {
+    it('sets the name attribute', () => {
+      render(<Textarea {...requiredProps} name="test-name" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('name', 'test-name');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Textarea {...requiredProps} name="unique-name" ref={ref} />);
+      expect(ref.current.name).toBe('unique-name');
+    });
+  });
+
+  describe('onBlur API', () => {
+    it('sets the onblur handler', () => {
+      const onBlurSpy = jest.fn();
+      render(<Textarea {...requiredProps} onBlur={onBlurSpy} />);
+      fireEvent.blur(screen.getByLabelText('the label'));
+      expect(onBlurSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onChange API', () => {
+    it('sets the onchange handler', () => {
+      const onChangeSpy = jest.fn();
+      render(<Textarea {...requiredProps} onChange={onChangeSpy} />);
+      fireEvent.change(screen.getByLabelText('the label'), { target: { value: 'new value' } });
```

Comment:
We prefer `userEvent interactions. Something like:

```suggestion
      userEvent.type(screen.getByLabelText('the label'), 'new value');
```